### PR TITLE
[GH-9598] Improve autocomplete highlighting when using mouse and keyboard together

### PR DESCRIPTION
**Pull request by larkox**
#### Summary

Improve autocomplete highlighting when using mouse and keyboard together
#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/9598
#### Related Pull Requests

None

#### Screenshots

None

### CODE REVIEW

**Comment by larkox** [comment]
I have done not much more as getting the original PR by @s4kh, merge it with master, and correct one of the latest comments he had on the PR, so most of the credit so far should go to @s4kh .

**Comment by lindy65** [comment]
Hi @larkox,

Thanks for the PR! `~` and `@` suggestion list navigation look good to me in:
- center channel and RHS message input boxes
- search `in:` and `from:`
- CTRL/CMD+K channel switcher

1. However, when using DIRECT MESSAGES `More...` modal I see too highlights - one from keyboard arrows and one from mouse hover over.  Not sure whether this is within the scope of this PR?

image

2. I also see the double highlight for mouse hover and keyboard arrow use in the Main Menu > Invite People > Add or Invite People modal. Also not sure whether this is within the scope of this PR?

add or invite people

Let me know whether these can be addressed in this PR or whether a separate Jira ticket should be opened for them. Thanks!

**Comment by larkox** [comment]
Thanks @lindy65 ! Since I was working on top of a previous PR and I have not yet enough control of mattermost as a whole, I missed those cases. I am pretty sure they are in the scope of this PR, so I will get my hands on this as soon as I can.
Thank you for pointing those out!

**Comment by lindy65** [comment]
Fantastic! Thanks @larkox - please let me know once ready to retest 👍

**Comment by larkox** [comment]
Thanks @lindy65 ! This should be already ready to review. There are many places where this can be affecting. I think I covered all the cases, but I would not be surprised if there is still some menu or some case I have not covered. Please let me know if you find any problem!

File: components/multiselect/multiselect_list.tsx [hunk 1]
```diff
@@ -137,12 +138,27 @@ export default class MultiSelectList extends React.Component<Props, State> {
...
         );
     }
 
+    private onMouseMove = (option: Value) => {
+        let i;
+        for (i = 0; i < this.props.options.length; i++) {
+            if (option === this.props.options[i]) {
+                break;
+            }
+        }
```
Comment (Willyfrog) [inline components/multiselect/multiselect_list.tsx]: wouldn't [indexOf](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf) be a more concise way for this?

**Comment by Willyfrog** [comment]
@larkox can you fix the conflicts so I can merge?

**Pull request merged**